components/bitbucket/new-commit.js

```diff
@@ -82,74 +59,75 @@ module.exports = {
     doesEventContainNewCommits(change) {
       return change.commits && change.commits.length > 0;
     },
+    getBaseCommitHash(change) {
+      return _.get(change, [
+        "old",
+        "target",
+        "hash",
+      ]);
+    },
     generateMeta(commit) {
       const {
         hash,
         message,
         date,
       } = commit;
-      const summary = `New commit: ${message} (${hash})`;
+      const commitTitle = message
+        .split("\n")
+        .shift();
+      const summary = `New commit: ${commitTitle} (${hash})`;
       const ts = +new Date(date);
       return {
         id: hash,
         summary,
         ts,
       };
     },
-  },
-  async run(event) {
-    const { headers, body } = event;
-
-    // Reject any calls not made by the proper BitBucket webhook.
-    if (!this.bitbucket.isValidSource(headers, this.db)) {
-      this.http.respond({
-        status: 404,
-      });
-      return;
-    }
-
-    // Acknowledge the event back to BitBucket.
-    this.http.respond({
-      status: 200,
-    });
+    async processEvent(event) {
+      const { body } = event;
+      const { changes = [] } = body.push;
 
-    // Push events can be for different branches, tags and
-    // causes. We need to make sure that we're only processing events
-    // that are related to new commits in the particular branch
-    // that the user indicated.
-    const { changes = [] } = body.push;
-    const isEventRelevant = changes
-      .filter(this.isEventForThisBranch)
-      .filter(this.doesEventContainNewCommits)
-      .length > 0;
-    if (isEventRelevant) {
-      const lastProcessedCommitHash = this.db.get("lastProcessedCommitHash");
-      const allCommits = this.bitbucket.getCommits({
-        workspaceId: this.workspaceId,
-        repositoryId: this.repositoryId,
-        branchName: this.branchName,
-        lastProcessedCommitHash,
-      });
+      // Push events can be for different branches, tags and
+      // causes. We need to make sure that we're only processing events
+      // that are related to new commits in the particular branch
+      // that the user indicated.
+      const eventsForThisBranch = changes.filter(this.isEventForThisBranch);
+      const isEventRelevant = eventsForThisBranch
+        .filter(this.doesEventContainNewCommits)
+        .length > 0;
+      if (isEventRelevant) {
```

Comment:
Since we use golden-path style elsewhere, consider negating this condition and returning early if the event isn't relevant.

Reply:
Noted, will do.